### PR TITLE
Ignore ApplicationExperience#application_form_id

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

The application_experience table should have all the experienceable
columns populated meaning we can rely on those columns to link it to the
application_form.

This commit ignores the application_form_id, another commit will remove
this column.

We also need to remove the data migrations that use this column.

And we need to explicitly delete the application work experiences in
`delete_test_applications.rb` because having polymorphic associations
will not delete the work experiences when deleting the candidate, it
doesn't do cascade deleting

Same issue in the candidate_spec file

## Changes proposed in this pull request

Ignore application_form_id
Remove data migrations that use this, they don't need to be run anymore
Update specs

## Guidance to review

Create/edit work experiences for a candidate on review or local

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [x] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Edit work experiences - Change db structure to accommodate work experience edits](https://trello.com/c/VzPy9EWy/2163-edit-work-experiences-change-db-structure-to-accommodate-work-experience-edits)

**Review by elceebee — APPROVED** [review]
I've tested this on the review app and it seems to work fine. 

This whole piece of work has been so intricate. I'm sure it has required a lot of patience and planning. Very impressive.

**Comment by CatalinVoineag** [comment]
> I've tested this on the review app and it seems to work fine.
> 
> This whole piece of work has been so intricate. I'm sure it has required a lot of patience and planning. Very impressive.

Team effort, this helped a lot https://dcyoung.dev/shorts/make-an-existing-relationship-polymorphic

File: app/models/candidate.rb [hunk 1]
```diff
@@ -78,6 +78,16 @@ def pseudonymised_id
     Digest::SHA2.hexdigest(id.to_s)
   end
 
+  def delete
+    application_form_ids = application_forms.pluck(:id)
```
Comment (dcyoung-dev) [inline app/models/candidate.rb:82]: Could we handle this with a `dependant: :destroy` on the `has_many: :application_work_experiences`?

Comment (CatalinVoineag) [reply app/models/candidate.rb:82]: It doesn't seem to do the job `dependent: destroy or delete` I think it doesn't work because it's a polymorphic link

**Review by dcyoung-dev — APPROVED** [review]

**Pull request merged**